src/scene_debug.cpp

```diff
@@ -102,78 +205,105 @@ void Scene_Debug::Update() {
+						numberinput_window->SetVisible(true);
+						numberinput_window->SetActive(true);
+						numberinput_window->SetMaxDigits(7);
+						numberinput_window->Refresh();
+					}
+					break;
+				case eItem:
+					if (GetIndex() <= Data::items.size()) {
+						var_window->SetActive(false);
+						numberinput_window->SetNumber(Main_Data::game_party->GetItemCount(GetIndex()));
+						numberinput_window->SetShowOperator(false);
+						numberinput_window->SetVisible(true);
+						numberinput_window->SetActive(true);
+						numberinput_window->SetMaxDigits(2);
+						numberinput_window->Refresh();
+					}
+					break;
+				case eBattle:
+					if (GetIndex() <= Data::troops.size()) {
+						Scene::PopUntil(Scene::Map);
+						if (Scene::instance) {
+							Game_Character *player = Main_Data::game_player.get();
+							Game_Battle::SetTerrainId(Game_Map::GetTerrainTag(player->GetX(), player->GetY()));
+							Game_Map::SetupBattle();
+							Game_Temp::battle_troop_id = GetIndex();
+							Game_Temp::battle_formation = 0;
+							Game_Temp::battle_escape_mode = 2;
+							Game_Temp::battle_defeat_mode = 1;
+							Game_Temp::battle_first_strike = 0;
+							Game_Temp::battle_result = Game_Temp::BattleVictory;
+							Game_Battle::SetBattleMode(0);
+							Game_Temp::battle_calling = true;
+							Game_Temp::transition_menu = false;
+							static_cast<Scene_Map*>(Scene::instance.get())->CallBattle();
+						}
+					}
+					break;
+				default:
+					break;
 			}
 			var_window->Refresh();
 		} else if (numberinput_window->GetActive()) {
-			Game_Variables[GetIndex()] = numberinput_window->GetNumber();
+			if (mode == eVariable) {
+				Game_Variables[GetIndex()] = numberinput_window->GetNumber();
+			} else if (mode == eGold) {
+				auto delta = numberinput_window->GetNumber() - Main_Data::game_party->GetGold();
+				Main_Data::game_party->GainGold(delta);
+				range_index = 4;
+			} else if (mode == eItem) {
+				auto delta = numberinput_window->GetNumber() - Main_Data::game_party->GetItemCount(GetIndex());
+				Main_Data::game_party->AddItem(GetIndex(), delta);
+			}
 			numberinput_window->SetActive(false);
 			numberinput_window->SetVisible(false);
-			var_window->SetActive(true);
-			var_window->Refresh();
+			if (var_window->GetVisible()) {
+				var_window->SetActive(true);
+				var_window->Refresh();
+			} else {
+				mode = eMain;
+				range_window->SetIndex(4);
+				range_window->SetActive(true);
+				UpdateRangeListWindow();
+			}
 		}
 		Game_Map::SetNeedRefresh(Game_Map::Refresh_All);
-	} else if (range_window->GetActive() &&  Input::IsTriggered(Input::RIGHT)) {
-		if (current_var_type != TypeGeneral) {
-			range_page++;
-		}
-		if (current_var_type == TypeSwitch && !Game_Switches.IsValid(range_page*100+1)) {
-			range_page = 0;
-			current_var_type++;
-		} else if (current_var_type == TypeInt && !Game_Variables.IsValid(range_page*100+1)) {
+	} else if (range_window->GetActive() && Input::IsTriggered(Input::RIGHT)) {
```

Comment:
you could use IsRepeated here (Same for Input::LEFT)

Reply:
Thats in #1439